### PR TITLE
Fix to properly add the tracked elements

### LINKED ISSUE #61: WithTracking feature does not appear to work
On the latest code on main, when using ElementQueries.withTracking = true, the instance method detach() doesn't detach all the elements that it's supposed to track because the tracked elements array is empty.

Steps to reproduce:
1.  set ElementQueries.withTracking = true;
2.  call ElementQueries.init();
3.  at a later point, call ElementQueries.instance.detach();
4.  Notice that in detach() the elements[] is empty (line 274)

Expected behavior:
When configured withTracking = true, calling ElementQueries.instance.detach() should remove all tracked elements.

### PR DESCRIPTION
**Pull request by rhuynh**
This is a fix to properly add the tracked elements to the elements array.  Previously, this.withTracking would end up referencing window instead of the ElementQueries instance.  The elements.indexOf(element) is to prevent adding duplicate elements.

Fixes #61

### CODE REVIEW

**Comment by marcj** [comment]
thanks!
